Rock/Workflow/Action/Communications/SendEmail.cs

```diff
@@ -176,6 +183,16 @@ public override bool Execute( RockContext rockContext, WorkflowAction action, Ob
                                         action.AddLogEntry( "Invalid Recipient: No valid group id or Guid", true );
                                     }
 
+
+                                    if ( groupRoleValueGuid.HasValue )
+                                    {
+                                        GroupTypeRole groupTypeRole = new GroupTypeRoleService( rockContext ).Get( groupRoleValueGuid.Value );
+                                        if ( groupTypeRole != null )
+                                        {
+                                            qry = qry.Where( m => m.GroupRoleId == groupTypeRole.Id );
```

Comment:
Instead of doing the query to GroupTypeRoleService a couple lines above, you can just modify this Where clause to use the Guid value...
```
    qry = qry.Where( m => m.GroupRole != null && m.GroupRole.Guid.Equals( groupRoleValueGuid.Value ) );
```